### DISCUSSION TITLE
Seperate popup menu for each treeItem

### DISCUSSION THREAD

**Discussion by peter-scholtens** [fltk-rs/fltk-rs#1296] [General]
I try to implement a different context pull down menu for each item in a tree, similar to the method demonstrated in [menubutton.rs](https://github.com/fltk-rs/fltk-rs/blob/master/fltk/examples/menubutton.rs).
```
let tree_item = tree.add(&label).unwrap();
let mut menu = menu::MenuButton::default()
    .size_of(&tree_item)
    .center_of(&tree_item)
    .with_type(menu::MenuButtonType::Popup3);
```
It fails however with:
> the trait `fltk::prelude::WidgetExt` is not implemented for `TreeItem`

Changing `.size_of(&tree_item)` to `.size_of(&tree)` and `.center_of(& ... )` too works, but creates the same popup for the whole tree. What's the best way to get individual popup menus for each TreeItem?

**Comment by MoAlyousef** [comment]
You might be able to call MenuButton::popup() inside the tree's handle method (since otherwise the tree doesn't respond to right clicks):
```rust
use fltk::{enums::*, prelude::*, *};

fn main() {
    let app = app::App::default();
    let mut win = window::Window::default().with_size(800, 600);
    let mut tree = tree::Tree::new(0, 0, 800, 600, None);
    tree.add("Hello");
    tree.add("again");
    let mut mb = menu::MenuButton::default().with_type(menu::MenuButtonType::Popup3);
    mb.add_choice("Some info");
    mb.set_frame(FrameType::FlatBox);
    tree.handle(move |t, e| match e {
        Event::Push => {
            if app::event_mouse_button() == app::MouseButton::Right {
                if let Some(i) = t.first_selected_item() {
                    mb.popup();
                }
                true
            } else {
                false
            }
        }
        _ => false,
    });
    win.end();
    win.show();

    app.run().unwrap();
}
```

**Reply by peter-scholtens** [reply]
Clear!
I only want to avoid the first click: in the above case user has to _first_ select the item and then _secondly_ right click. But I think the functions `fltk::app::belowmouse` and `tree::method.set_item_focus` allow me to do so, similar as Gnome's Nautilus file browser does..
